sway-lib-std/src/inputs.sw

```diff
@@ -156,3 +175,57 @@ pub fn input_count() -> u8 {
         },
     }
 }
+
+////////////////////////////////////////
+// Message Inputs
+////////////////////////////////////////
+
+/// Get the message id of the input message at `index`.
+pub fn input_message_msg_id(index: u64) -> b256 {
+    __gtf::<b256>(index, GTF_INPUT_MESSAGE_MESSAGE_ID)
```

Comment:
Should the functions be checking that the input type is actually a message? I'm not sure if we should return `Option::None` (like `input_predicate_data_pointer`) or just `revert(0)` (like `input_predicate_data`). I'm kinda leaning towards a revert

Reply:
Yeah, it could probably go either way here...
There are a bunch of these functions using `__gtf` that only make sense for specific types of inputs/outputs, and only some check the type IIRC.
I'll need to review for consistency and probably refactor a bit.

Reply:
I've made a change here by implementing `Eq` for `Input`, and then adding a function to validate an Input type which can be reused:

```
impl Eq for Input {
    fn eq(self, other: Self) -> bool {
        match (self, other) {
            (Input::Coin, Input::Coin) => true,
            (Input::Contract, Input::Contract) => true,
            (Input::Message, Input::Message) => true,
            _ => false,
        }
    }
}

fn valid_input_type(index: u64, expected_type: Input) -> bool {
    let input_type: Input = input_type(index);
    input_type == expected_type
}

/// Get the message id of the input message at `index`.
pub fn input_message_msg_id(index: u64) -> b256 {
    assert(valid_input_type(index, Input::Message));
    __gtf::<b256>(index, GTF_INPUT_MESSAGE_MESSAGE_ID)
}
```